targets/TARGET_STM/can_api.c

```diff
@@ -646,8 +646,12 @@ void can_irq_set(can_t *obj, CanIrqType type, uint32_t enable)
 #include <string.h>
 #include <inttypes.h>
 
+#define ENABLED true
```

Comment:
Rather than inventing your own values, just name the variable so booleans work:

    static bool rx_irq_enabled = false;